### ISSUE TITLE
Launching app options dialog causes exception when dmf_control_board plugin is enabled (Trac #102)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#106]
App options dialog fails to launch with:

```
(<type 'exceptions.TypeError'>, TypeError("'Form' dictionary type has no fields defined",), <traceback object at 0x9eeb8c4>) {}
  File "/home/christian/Documents/dev/udrop/microdrop/microdrop/gui/main_window_controller.py", line 191, in on_menu_app_options_activate
    AppOptionsController().run()
  File "/home/christian/Documents/dev/udrop/microdrop/microdrop/gui/app_options_controller.py", line 93, in run
    self.form_views[name] = FormView()
  File "/home/christian/local/pylib/pygtkhelpers/delegates.py", line 85, in __init__
    self.create_ui()
  File "/home/christian/local/pylib/pygtkhelpers/forms.py", line 120, in create_ui
    self.form = FieldSet(self, self.schema_type)
  File "/home/christian/local/pylib/pygtkhelpers/forms.py", line 81, in __init__
    self.schema = schema_type()
  File "/home/christian/Documents/dev/udrop/microdrop/microdrop/flatland/schema/containers.py", line 742, in __init__
    type(self).__name__))
```

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/102
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "App options dialog fails to launch with:\n\n{{{\n(<type 'exceptions.TypeError'>, TypeError(\"'Form' dictionary type has no fields defined\",), <traceback object at 0x9eeb8c4>) {}\n  File \"/home/christian/Documents/dev/udrop/microdrop/microdrop/gui/main_window_controller.py\", line 191, in on_menu_app_options_activate\n    AppOptionsController().run()\n  File \"/home/christian/Documents/dev/udrop/microdrop/microdrop/gui/app_options_controller.py\", line 93, in run\n    self.form_views[name] = FormView()\n  File \"/home/christian/local/pylib/pygtkhelpers/delegates.py\", line 85, in __init__\n    self.create_ui()\n  File \"/home/christian/local/pylib/pygtkhelpers/forms.py\", line 120, in create_ui\n    self.form = FieldSet(self, self.schema_type)\n  File \"/home/christian/local/pylib/pygtkhelpers/forms.py\", line 81, in __init__\n    self.schema = schema_type()\n  File \"/home/christian/Documents/dev/udrop/microdrop/microdrop/flatland/schema/containers.py\", line 742, in __init__\n    type(self).__name__))\n}}}", 
    "reporter": "cfobel", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Launching app options dialog causes exception when dmf_control_board plugin is enabled", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-04-17T23:37:10", 
    "milestone": "Microdrop 1.0", 
    "owner": "cfobel", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120417T23:42:03: 
* christian@fobel.net commented:

> commit 0f07df2cb0edb88a0ea19261a55a40a70ca18a5d
> Author: Christian Fobel <christian@fobel.net>
> Date:   Tue Apr 17 19:32:18 2012 -0400
> 
>     Fix #106, bug with dmf_control_board app fields
>     
>     All dmf_control_board (v0.1-216-g03278e8) app fields have 'show_in_gui' set to
>     False.  This causes the app options controller to fail when generating a form
>     view for each plugin with app options.  Modify the app options controller to
>     ignore forms with where all fields are set to 'show_in_gui=False'.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "" to "microdrop"
